internal/provider/iam_workload_identity_binding.go

```diff
@@ -0,0 +1,384 @@
+				MarkdownDescription: "Description of the binding.",
+				Optional:            true,
+				Computed:            true,
+				Default:             stringdefault.StaticString("Created by Terraform"),
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"roles": schema.ListAttribute{
+				MarkdownDescription: "List of roles associated with the binding (e.g. agent mrn).",
+				ElementType:         types.StringType,
+				Optional:            true,
+				Computed:            true,
+				PlanModifiers: []planmodifier.List{
+					listplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"issuer_uri": schema.StringAttribute{
+				MarkdownDescription: "URI for the token issuer, e.g. https://accounts.google.com.",
+				Required:            true,
+			},
+			"subject": schema.StringAttribute{
+				MarkdownDescription: "Unique identifier to confirm.",
+				Required:            true,
+			},
+			"expiration": schema.Int32Attribute{
+				MarkdownDescription: "Expiration in seconds associated with the binding.",
+				Optional:            true,
+			},
+			"allowed_audiences": schema.ListAttribute{
+				MarkdownDescription: " List of allowed audiences.",
+				ElementType:         types.StringType,
+				Optional:            true,
+				Computed:            true,
+				PlanModifiers: []planmodifier.List{
+					listplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"mappings": schema.MapAttribute{
+				MarkdownDescription: "List of additional configurations to confirm.",
+				Optional:            true,
+				ElementType:         types.StringType,
+				PlanModifiers: []planmodifier.Map{
+					mapplanmodifier.UseStateForUnknown(),
+				},
+			},
+		},
+	}
+}
+
+func (r *IAMWorkloadIdentityBindingResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+
+	client, ok := req.ProviderData.(*ExtendedGqlClient)
+
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected *http.Client. Got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
+
+	r.client = client
+}
+
+type WIFAuthBinding struct {
+	Mrn              string
+	Name             string
+	Description      string
+	Scope            string
+	Roles            []string
+	Expiration       int32
+	IssuerURI        string `graphql:"issuerURI"`
+	Subject          []byte
+	Mappings         []KeyValue
+	AllowedAudiences []string
+}
+type WIFExternalAuthConfig struct {
+	UniverseDomain   string `graphql:"universe_domain"`
+	Type             string
+	Audience         string
+	SubjectTokenType string `graphql:"subject_token_type"`
+	Scopes           []string
+	IssuerURI        string `graphql:"issuer_uri"`
+}
+
+// Workaround because of inconsistencies.
+type CreateWIFAuthBindingInput struct {
```

Comment:
Issue URI is not consistent in the Input and Payload of the create mutation:
 
* Input `issueUri` 
* Payload `issueURI` 
* Payload 2 `issue_uri`

We should probably fix this before merging this PR.